### PR TITLE
Movie-Info-App

### PR DESCRIPTION
**Pull request by AckermanLevi1**
# Title and Issue number 

Title :

Issue No. :#215

Code Stack : HTML, CSS, JAVASCRIPT

Close #<215>

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the Index.html file for my contribution

<h1>Website Link</h1>
<a href="https://moviestadka.netlify.app/">Click here</a>

### CODE REVIEW

File: native/machine-coding/Movie-Info-App/index.css [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    margin: 0;
+    padding: 0;
+}
+
+/* Set background image and styles for the body */
+body {
+    background-image: url('https://img.freepik.com/free-photo/assortment-cinema-elements-red-background-with-copy-space_23-2148457848.jpg?w=1380&t=st=1697217223~exp=1697217823~hmac=fb98e196d954c9a841de136c18083f1a3d2a7059878372928e8628e676225272');
+    background-size: cover;
+    background-position: center;
+    font-family: Arial, sans-serif;
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.css:12]: dont add any font-family

File: native/machine-coding/Movie-Info-App/index.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <script src="./index.js" defer></script>
+    <script src="../../helpers/header.js" type="module"></script>
+    <link rel="stylesheet" href="./index.css">
+    <title>Movie Info Website</title>
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.html:9]: only these are enough in head

```
<script src="./index.js" defer></script>
<script src="../../helpers/header.js" type="module"></script>
<link rel="stylesheet" href="./index.css">
```

**Comment by arpansaha13** [comment]
You do not need to create a new PR after every change
When you push a new commit in your branch (same branchfrom where PR was created) they will automatically appear here

File: native/machine-coding/Movie-Info-App/index.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+document.addEventListener('DOMContentLoaded', function() {
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.js:1]: as defer is used while loading script, I don't think we need the DOMContentLoaded listener. Can be written directly

File: native/machine-coding/Movie-Info-App/index.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        fetch(`https://www.omdbapi.com/?t=${query}&apikey=${apiKey}`)
+            .then(response => response.json())
+            .then(data => {
+                const movieTitle = document.getElementById('movie-title');
+                const movieGenre = document.getElementById('movie-genre');
+                const movieCast = document.getElementById('movie-cast');
+                const movieRating = document.getElementById('movie-rating');
+                const movieReleased = document.getElementById('movie-released');
+                const movieDirector = document.getElementById('movie-director');
+                const moviePoster = document.getElementById('movie-poster');
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.js:25]: can we not move these query selectors outside to the top of file and use the variables inside

File: native/machine-coding/Movie-Info-App/index.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    const movieInfoDiv = document.querySelector('.movie-info');
+    movieInfoDiv.style.display = 'none';
+
+    const searchButton = document.getElementById('search-button');
+    const searchInput = document.getElementById('search-input');
+
+    searchButton.addEventListener('click', () => {
+        // Fetch movie data from an API or your database and update the DOM elements accordingly.
+        // For example, you can use the OMDB API for this purpose.
+        const apiKey = 'adff2bf8';
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.js:12]: move to the top of file

File: native/machine-coding/Movie-Info-App/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                const movieDirector = document.getElementById('movie-director');
+                const moviePoster = document.getElementById('movie-poster');
+
+                movieTitle.textContent = data.Title;
+                movieGenre.textContent = data.Genre;
+                movieCast.textContent = data.Actors;
+                movieRating.textContent = data.imdbRating;
+                movieReleased.textContent = data.Released;
+                movieDirector.textContent = data.Director;
+                moviePoster.src = data.Poster;
```
Comment (sadanandpai) [inline native/machine-coding/Movie-Info-App/index.js:33]: can we move the assignment task to a separate function and keep the fetch lean for readability

File: native/helpers/challenges.js
Comment (sadanandpai) [inline native/helpers/challenges.js:1]: name the folder in small case only. (can omit **app**)

movie-Info

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by AckermanLevi1** [comment]
@sadanandpai pls check

File: native/helpers/challenges.js [hunk 7]
```diff
@@ -453,4 +453,11 @@ export const challenges = [
   { title: 'Emoji Editor', link: '#' },
   { title: 'Sort Table', link: '#' },
   { title: 'Infinite Scroll', link: '#' },
+  {
+    title: 'Movie-Info-App(online)',
+    link: 'Movie-Info-App',
```
Comment (sadanandpai) [inline native/helpers/challenges.js:458]: `link: 'movie-info'`

should be same as folder name

File: native/machine-coding/movie-info/index.html [hunk 8]
```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (sadanandpai) [inline native/machine-coding/movie-info/index.html:5]: above meta tags not needed

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by AckermanLevi1** [comment]
@sadanandpai pls check

**Comment by sadanandpai** [comment]
@AckermanLevi1 

Not handled for invalid movie
image

Also pressing enter is not working after entering movie name. 

Please work on those

**Pull request closed without merging**